### ISSUE TITLE
Global Styles: revisit order of blocks to improve information architecture

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#44840]
Similar to how the blocks in the inserter have been re-examined to weight and group them appropriately, I believe the same needs to be done with the Styles > Blocks interface:

<img width="288" alt="Screen Shot 2022-10-10 at 6 31 22 PM" src="https://user-images.githubusercontent.com/26996883/194962504-abb9e2c9-38e5-44b3-9309-a1dd6dfcb34b.png">

For example, it feels really odd that archives, audio, calendar, and categories list are in the visible section on desktop. What can we do to perhaps add categories to match what you see in the Inserter (have consistency) and weight what folks are more likely to adjust here? cc @WordPress/gutenberg-design  for thoughts.

**Comment by jasmussen** [comment]
Good issue. Search helps a great deal, prior to that I would've wanted alphabetization at the very least, as that seems like it would be the case in lists like these. I understand weighting is important. In that light, should Paragraph be so high? Isn't it better to change GS > Typography instead of just the paragraph block?

However could/should we combine categories with alphabetization? I.e. we could have "Common", "Layout", "Templates"? Alternately, we could duplicate the same categories the inserter has, for some recognition, perhaps with the added "Templates" category?

**Comment by annezazu** [comment]
For some reason, my brain goes to wanting categories, similar to what we see in the Inserter. That may be an old way of thinking though but, for now, it feels like it might help with consistency in experience.

**Comment by jameskoster** [comment]
Alphabetisation seems like a good middle-ground for now, with filtering by category being a further enhancement.

**Comment by apeatling** [comment]
I've tested and approved the PR above for this, @jasmussen / @annezazu / @jameskoster just a final check to make sure you're still good with this list changing to alphabetical before it's merged? I think it's an improvement, just a very noticeable one so good to get another thumbs up.

**Comment by annezazu** [comment]
@apeatling let's roll with it. Thank you for following up here!

**Comment by mtias** [comment]
I don't think we should do alphabetical. We used to have this initially and changed it because it's unusable as a sorting scheme when you have various libraries coexisting and you see obscure blocks at the top. The intention was to match how the inserter itself lists things.

**Comment by dsas** [comment]
The alphabetisation PR currently open (https://github.com/WordPress/gutenberg/pull/45678) lists all core blocks alphabetically, and then all non-core blocks alphabetically. I think it's an improvement over what currently exists but it isn't the ideal end state. If someone is about to open a PR for the ideal end state then lets just drop https://github.com/WordPress/gutenberg/pull/45678, otherwise I think it should be merged (or have been merged months ago :)) to improve things until the ideal end state is ready.

**Comment by annezazu** [comment]
I also think the Inserter listing needs to be improved when you're in the Site Editor so perhaps these two need to be addressed in tandem: https://github.com/WordPress/gutenberg/issues/42480

**Comment by richtabor** [comment]
> The intention was to match how the inserter itself lists things.

Just a note that this Styles > Blocks list isn't the ideal experience for finding and styling blocks — that's where the [Style Book](https://github.com/WordPress/gutenberg/pull/45960) comes into play. Where you can click on a block in the Style Book and have its panel open up for editing in the Global Styles sidebar. 

That saying, I'm fine with pulling the same sorting that we use in the inserter — though I'd say this is low priority.

**Comment by mtias** [comment]
We should also allow selecting any block when GS is open and do a deep selection of the block types panel, not just in Style Book.

**Comment by ndiego** [comment]
Hey team, just checking in as this issue is on the [6.3 Project Board](https://github.com/orgs/WordPress/projects/103/views/1), and I do not see any recent activity. Are we still aiming for 6.3 with this? Otherwise, I will remove it from the board. Thanks!
